app/lib/features/tasks/pages/tasks_list_page.dart

```diff
@@ -0,0 +1,215 @@
+import 'package:acter/features/search/providers/search.dart';
+
+class TasksListPage extends ConsumerStatefulWidget {
+  static const scrollView = Key('space-task-lists');
+  static const createNewTaskListKey = Key('tasks-create-list');
+  static const taskListsKey = Key('tasks-task-lists');
+  final String? spaceId;
+
+  const TasksListPage({super.key, this.spaceId});
+
+  @override
+  ConsumerState<TasksListPage> createState() => _TasksListPageConsumerState();
+}
+
+class _TasksListPageConsumerState extends ConsumerState<TasksListPage> {
+  final TextEditingController searchTextController = TextEditingController();
+
+  String get searchValue => ref.watch(searchValueProvider);
+  final ValueNotifier<bool> showCompletedTask = ValueNotifier(false);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: _buildAppBar(),
+      body: _buildBody(),
+    );
+  }
+
+  AppBar _buildAppBar() {
+    return AppBar(
+      centerTitle: false,
+      title: Column(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(L10n.of(context).tasks),
+          if (widget.spaceId != null) _buildSpaceName(),
+        ],
+      ),
+      actions: [
+        ValueListenableBuilder(
+          valueListenable: showCompletedTask,
+          builder: (context, value, child) {
+            return TextButton.icon(
+              onPressed: () => showCompletedTask.value = !value,
+              icon: Icon(
+                value
+                    ? Icons.visibility_off_outlined
+                    : Icons.visibility_outlined,
+                size: 18,
+              ),
+              label: Text(
+                value
+                    ? L10n.of(context).hideCompleted
+                    : L10n.of(context).showCompleted,
+              ),
+              style: OutlinedButton.styleFrom(
+                padding: const EdgeInsets.symmetric(horizontal: 10),
+              ),
+            );
+          },
+        ),
+        AddButtonWithCanPermission(
+          key: TasksListPage.createNewTaskListKey,
+          canString: 'CanPostTaskList',
+          onPressed: () => showCreateUpdateTaskListBottomSheet(
+            context,
+            initialSelectedSpace: widget.spaceId,
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildSpaceName() {
+    String spaceName =
+        ref.watch(roomDisplayNameProvider(widget.spaceId!)).valueOrNull ?? '';
+    return Text(
+      '($spaceName)',
+      overflow: TextOverflow.ellipsis,
+      style: Theme.of(context).textTheme.labelLarge,
+    );
+  }
+
+  Widget _buildBody() {
+    AsyncValue<List<TaskList>> tasksList;
+
+    if (searchValue.isNotEmpty) {
+      tasksList = ref.watch(
+        tasksListSearchProvider(
+          (spaceId: widget.spaceId, searchText: searchValue),
+        ),
+      );
+    } else {
+      tasksList = ref.watch(tasksListsProvider(widget.spaceId));
+    }
```

Comment:
we could also handle that by just having the entire list returned if `searchValue.isEmpty` on the search provider ... making this more straight forward and moving the "logic" into the provider (where logic ideally resides)